RecoTracker/MkFit/plugins/MkFitEventOfHitsProducer.cc

```diff
@@ -85,14 +87,52 @@ void MkFitEventOfHitsProducer::produce(edm::StreamID iID, edm::Event& iEvent, co
     const auto& trackerGeom = iSetup.getData(geomToken_);
     const auto& badStrips = siStripQuality.getBadComponentList();
     for (const auto& bs : badStrips) {
-      const auto& surf = trackerGeom.idToDet(DetId(bs.detid))->surface();
       const DetId detid(bs.detid);
+      const auto& surf = trackerGeom.idToDet(detid)->surface();
       bool isBarrel = (mkFitGeom.topology()->side(detid) == static_cast<unsigned>(TrackerDetSide::Barrel));
       const auto ilay = mkFitGeom.mkFitLayerNumber(detid);
-      deadvectors[ilay].push_back({surf.phiSpan().first,
-                                   surf.phiSpan().second,
-                                   (isBarrel ? surf.zSpan().first : surf.rSpan().first),
-                                   (isBarrel ? surf.zSpan().second : surf.rSpan().second)});
+      const auto q1 = isBarrel ? surf.zSpan().first : surf.rSpan().first;
+      const auto q2 = isBarrel ? surf.zSpan().second : surf.rSpan().second;
+      if (bs.BadModule)
+        deadvectors[ilay].push_back({surf.phiSpan().first, surf.phiSpan().second, q1, q2});
+      else {  //assume that BadApvs are filled in sync with BadFibers
+        auto const& topo = dynamic_cast<const StripTopology&>(trackerGeom.idToDet(detid)->topology());
+        int firstApv = -1;
+        int lastApv = -1;
+
+        auto addRangeAPV = [&topo, &surf, &q1, &q2](int first, int last, mkfit::DeadVec& dv) {
+          auto const firstPoint = surf.toGlobal(topo.localPosition(first * 128));
+          auto const lastPoint = surf.toGlobal(topo.localPosition((last + 1) * 128));
+          float phi1 = firstPoint.phi();
+          float phi2 = lastPoint.phi();
+          if (reco::deltaPhi(phi1, phi2) > 0)
+            std::swap(phi1, phi2);
+          LogTrace("SiStripBadComponents")
+              << "insert bad range " << first << " to " << last << " " << phi1 << " " << phi2;
+          dv.push_back({phi1, phi2, q1, q2});
+        };
+
+        for (int apv = 0; apv < 6; ++apv) {
```

Comment:
might the apv max number be a named constant?

Reply:
@pieterdavid @mmusich 
do we have max APVs and n strips per APV formalized in common constants somewhere?
I tried to git grep, but all that I found was hardcoded 6 and 128.

Reply:
>  do we have max APVs and n strips per APV formalized in common constants somewhere?

they are (at least) in this file: https://github.com/cms-sw/cmssw/blob/master/CalibTracker/SiStripCommon/data/SiStripDetInfo.dat.
Or you can do something like:
```C++
      unsigned short nApvs = stripDet->specificTopology().nstrips() / 128;
```
128 is univeral for all modules, I am not sure there's a constant somewhere.

Reply:
the constants are also here: https://github.com/cms-sw/cmssw/blob/master/DataFormats/SiStripCommon/interface/ConstantsForHardwareSystems.h#L38-L43 (6 as max number of APVs indirectly, as 2 per optical link * max 3 of these per module)

Reply:
> what about modules with less than 6 APVs?

it shouldn't matter here, if the `BadApvs` is filled correctly.

Reply:
Updated in commit 91d1cafa99953018fe8d4b3223eee72c437cce9c